### PR TITLE
Use assertJ in tests

### PR DESCRIPTION
**Pull request by mabdinur**
Use assertJ assertions instead of junit5

### CODE REVIEW

**Comment by jkwatson** [comment]
Rather than adding a few assertJ tests, I'd rather continue to use Truth until we convert everything at once. I definitely don't want to have both Truth and assertJ at the same time.

**Comment by anuraaga** [comment]
This is the `opentelemetry-java-instrumentation` repo we don't have any Truth here yet. Since Spring uses assertj natively (we don't even have to import it since it's included in their test starter) I'd like to keep consistency with Spring for these modules rather than SDK project here if that sounds OK.

File: instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java [hunk 1]
```diff
@@ -41,45 +40,39 @@ public Tracer customTestTracer() {
...
         .run(
             (context) -> {
-              assertTrue(
-                  "Application Context contains customTestTracer bean",
-                  context.containsBean("customTestTracer"));
-
-              assertFalse(
-                  "Application Context DOES NOT contain the otelTracer bean defined in TracerAutoConfiguration",
-                  context.containsBean("otelTracer"));
+              assertThat(context.containsBean("customTestTracer")).isEqualTo(true);
```
Comment (anuraaga) [inline instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java:50]:
```suggestion
              assertThat(context.containsBean("customTestTracer")).isTrue();
```

File: instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java [hunk 2]
```diff
@@ -41,45 +40,39 @@ public Tracer customTestTracer() {
...
+              assertThat(context.containsBean("otelTracer")).isEqualTo(false);
```
Comment (anuraaga) [inline instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java:51]:
```suggestion
              assertThat(context.containsBean("otelTracer")).isFalse();
```

File: instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java [hunk 3]
```diff
@@ -41,45 +40,39 @@ public Tracer customTestTracer() {
...
+  @DisplayName("when Application Context DOES NOT contain Tracer bean should initialize otelTracer")
+  void initalizeTracer() {
     this.contextRunner
         .withConfiguration(AutoConfigurations.of(TracerAutoConfiguration.class))
         .run(
             (context) -> {
-              assertTrue(
-                  "Application Context contains otelTracer bean",
-                  context.containsBean("otelTracer"));
+              assertThat(context.containsBean("otelTracer")).isEqualTo(true);
```
Comment (anuraaga) [inline instrumentation-core/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/TracerAutoConfigurationTest.java:62]:
```suggestion
              assertThat(context.containsBean("otelTracer")).isTrue();
```

**Comment by jkwatson** [comment]
oh yes...Very sorry. I got confused as to what repo I was looking at.  Never Mind Me.  ;)

**Pull request merged**